### PR TITLE
cleaning up code-base with any unused methods. 

### PR DESCRIPTION
**Pull request by shahzebsiddiqui**
The slurm and lsf configuration detection is not needed right now. Will add it later if needed.

### CODE REVIEW

**Comment by shahzebsiddiqui** [comment]
@vsoch I am just removing any unused methods from the framework. It should be relatively simple change.

**Comment by shahzebsiddiqui** [comment]
the second commit @vsoch  was to address this print issue i noticed. It comes up in several commands

```
Skipping module tree /usr/share/modulefiles/Linux because path does not exist
Skipping module tree /usr/share/modulefiles/Core because path does not exist
```

```
ssi29@ag-mxg-hulk090> buildtest config view
Skipping module tree /usr/share/modulefiles/Linux because path does not exist
Skipping module tree /usr/share/modulefiles/Core because path does not exist
BUILDTEST_MODULEPATH: []
EDITOR: vim
build:
  module:
    purge:
      force: false
  testdir: $HOME/.buildtest/testdir
module:
  list:
    exclude_version_files: true
    filter:
      include: []
    querylimit: -1
  loadtest:
    login: false
    numtest: -1
    purge_modules: true
  spider_view: current

```

```
ssi29@ag-mxg-hulk090> buildtest show --config
Skipping module tree /usr/share/modulefiles/Linux because path does not exist
Skipping module tree /usr/share/modulefiles/Core because path does not exist
BUILDTEST_MODULEPATH                               = /mxg-hpc/users/ssi29/easybuild-HMNS/modules/all/Core:/mxg-hpc/users/ssi29/spack/modules/linux-rhel7-x86_64/Core:/mxg-hpc/users/ssi29/easybuild/modules/all:/etc/modulefiles:/usr/share/modulefiles:/usr/share/lmod/lmod/modulefiles/Core
EDITOR                                             = vim
build[module][purge]                               = {'force': False}
build[testdir]                                     = /u/users/ssi29/.buildtest/testdir
module[list][exclude_version_files]                = True
module[list][filter]                               = {'include': []}
module[list][querylimit]                           = -1
module[loadtest][login]                            = False
module[loadtest][numtest]                          = -1
module[loadtest][purge_modules]                    = True
module[spider_view]                                = current

```

**Comment by shahzebsiddiqui** [comment]
the second commit @vsoch  was to address this print issue i noticed. It comes up in several commands

```
Skipping module tree /usr/share/modulefiles/Linux because path does not exist
Skipping module tree /usr/share/modulefiles/Core because path does not exist
```

```
ssi29@ag-mxg-hulk090> buildtest config view
Skipping module tree /usr/share/modulefiles/Linux because path does not exist
Skipping module tree /usr/share/modulefiles/Core because path does not exist
BUILDTEST_MODULEPATH: []
EDITOR: vim
build:
  module:
    purge:
      force: false
  testdir: $HOME/.buildtest/testdir
module:
  list:
    exclude_version_files: true
    filter:
      include: []
    querylimit: -1
  loadtest:
    login: false
    numtest: -1
    purge_modules: true
  spider_view: current

```

```
ssi29@ag-mxg-hulk090> buildtest show --config
Skipping module tree /usr/share/modulefiles/Linux because path does not exist
Skipping module tree /usr/share/modulefiles/Core because path does not exist
BUILDTEST_MODULEPATH                               = /mxg-hpc/users/ssi29/easybuild-HMNS/modules/all/Core:/mxg-hpc/users/ssi29/spack/modules/linux-rhel7-x86_64/Core:/mxg-hpc/users/ssi29/easybuild/modules/all:/etc/modulefiles:/usr/share/modulefiles:/usr/share/lmod/lmod/modulefiles/Core
EDITOR                                             = vim
build[module][purge]                               = {'force': False}
build[testdir]                                     = /u/users/ssi29/.buildtest/testdir
module[list][exclude_version_files]                = True
module[list][filter]                               = {'include': []}
module[list][querylimit]                           = -1
module[loadtest][login]                            = False
module[loadtest][numtest]                          = -1
module[loadtest][purge_modules]                    = True
module[spider_view]                                = current
```

File: buildtest/tools/menu.py [hunk 1]
```diff
@@ -95,9 +95,6 @@ def main_menu(self):
             version=f"""buildtest version {config_opts["BUILDTEST_VERSION"]}""",
         )
 
-    def get_parser(self):
-        return self.parser
-
```
Comment (vsoch) [inline buildtest/tools/menu.py:100]: +1 to remove this.

**Review by vsoch — APPROVED** [review]
These changes are very good - we're simplifying the library to not have extra modules / functions that aren't yet determined to be absolutely essential!

**Comment by shahzebsiddiqui** [comment]
@vsoch can you merge PR if it looks good.

**Comment by vsoch** [comment]
Yep, happy to! This improves the quality of the code base, so +1 to get it in the working devel branch.

**Pull request merged**